### PR TITLE
Fix #15322: Circus Music Not Playing

### LINKED ISSUE #15322: Circus does not have any sound
### Operating System

Ubuntu 20.04

### OpenRCT2 build

b2d0b54d135e1b82899e0e96affbf265c781179a

### Describe the issue

The sound effects from the Circus are completely gone.

It’s extremely likely this has been caused by the introduction of music objects, since the circus show was not included as one (it’s technically music, but behaves as a sound effect).

### Area(s) with issue?

_No response_

### Steps to reproduce

1. Let a Circus start its show
2. Listen for any sound

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by seanmajorpayne**
Added check to ride_update_music to see if ride type is circus. Separated circus music processing from the rest of the rides.

Previously, Circus would not play music.
Now, Circus will play music as expected.

Issue was just an 'if' statement that was returning immediately if the music flag for a ride was not set. As the circus is a sound effect, not music, it was continually returning.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
When was this mistake introduced. We don't add to the changelog unless the issue was in the previous release.

**Comment by seanmajorpayne** [comment]
I'm not sure. I'll double check the previous release.

**Comment by duncanspumpkin** [comment]
Although i really doubt there will be an issue with it we should increment the network version just in case. The music is ultimately used for some peep interaction i think.

**Comment by Gymnasiast** [comment]
@duncanspumpkin The bug has been introduced with the introduction of music objects. It is already present in 0.3.3. So it should be in the changelog.

**Comment by seanmajorpayne** [comment]
@duncanspumpkin Do you need me to update something in the code for the network? Or is that a deployment issue?

**Comment by duncanspumpkin** [comment]
See network base.cpp just increment the number in there. But leave that until review is completed as we may have modified it subsequently.

**Comment by seanmajorpayne** [comment]
Okay no problem.

**Comment by ocalhoun6** [comment]
> Although i really doubt there will be an issue with it we should increment the network version just in case. The music is ultimately used for some peep interaction i think.

Only the merry-go-round music and the fairground organ music, I think. So it shouldn't affect the circus.

**Comment by Broxzier** [comment]
Dodgems and one other soundtrack affect guest thoughts. Better safe than sorry.

File: src/openrct2/ride/Ride.cpp [hunk 1]
```diff
@@ -1814,7 +1817,6 @@ static void ride_music_update(Ride* ride)
             ride->music_tune_id = static_cast<uint8_t>(util_rand() % numTracks);
             ride->music_position = 0;
         }
-        return;
```
Comment (Gymnasiast) [inline src/openrct2/ride/Ride.cpp:1817]: Why did you remove this return?

File: src/openrct2/ride/Ride.cpp [hunk 2]
```diff
@@ -1803,9 +1806,9 @@ static void ride_music_update(Ride* ride)
         }
     }
 
-    // Select random tune from available tunes for a music style (of course only merry-go-rounds have more than one tune)
     if (ride->music_tune_id == 255)
     {
+        // Select random tune from available tunes for a music style (of course only merry-go-rounds have more than one tune)
```
Comment (Gymnasiast) [inline src/openrct2/ride/Ride.cpp:1811]: Stray change, could you revert this?

File: src/openrct2/ride/Ride.cpp (hunk 1 above)
Comment (seanmajorpayne) [reply src/openrct2/ride/Ride.cpp:1817]: Oh, that's strange. I didn't see this in Github Desktop when I reviewed.

I'll revert.

File: src/openrct2/ride/Ride.cpp (hunk 2 above)
Comment (seanmajorpayne) [reply src/openrct2/ride/Ride.cpp:1811]: Yes will do

File: src/openrct2/ride/Ride.cpp [hunk 3]
```diff
@@ -1754,22 +1754,25 @@ Staff* ride_get_assigned_mechanic(Ride* ride)
  */
 static void ride_music_update(Ride* ride)
 {
-    const auto& rtd = ride->GetRideTypeDescriptor();
```
Comment (Gymnasiast) [inline src/openrct2/ride/Ride.cpp:1757]: I think we could just remove this if altogether.

**Review by Gymnasiast — COMMENTED** [review]
Don’t change anything yet. I want to test some things.

**Comment by seanmajorpayne** [comment]
Okay no problem.

**Comment by Gymnasiast** [comment]
OK, I checked what I wanted. You can now proceed with the line notes.
You can also add yourself to contributors.md if you haven’t done so already.

**Comment by seanmajorpayne** [comment]
I've fixed those small issues. Should be ready for review/testing.

I added to contributors.md last PR, so all good there.

**Review by Gymnasiast — CHANGES_REQUESTED** [review]
Tested it. It doesn’t seem to work for existing circuses.

For example: open European Cultural Festival (WW), go to Main Circus Tent and listen. There is no circus music, even after the visitors exit and new ones enter.

**Comment by seanmajorpayne** [comment]
This does not appear to be a bug with the circus music directly, but something separate.

[See this line in RideAudio.cpp](https://github.com/OpenRCT2/OpenRCT2/blob/a5a63f839aca539ad75abf09a765e590e5ba973b/src/openrct2/ride/RideAudio.cpp#L194)

It is causing this log_warning.
```
object\ObjectManager.cpp:75 (ObjectManager::GetLoadedObject)]: Object index 65535 exceeds maximum for type 14.
```

I'm not sure what this means or why this would occur. Do you have any intuition of what is happening here?

**Comment by Gymnasiast** [comment]
That would be this issue: https://github.com/OpenRCT2/OpenRCT2/issues/15371

It’s caused by music being set to OBJECT_ENTRY_INDEX_NULL (=65535). I don’t know if it’s related to this, though.

**Comment by seanmajorpayne** [comment]
I believe the two are related. I am not able to replicate the circus sound issue you experienced on European Cultural Festival outside of that map. If you watch the console output and disable the circus, it should stop reporting that error. If you put it back on test or open, as soon as the music would start playing that error starts popping up again.

**Comment by spacek531** [comment]
I could not recreate the spam by building a new circus. Are circuses being imported incorrectly?

**Comment by seanmajorpayne** [comment]
What do you mean you could not recreate the spam? If you are talking about
console I only saw that for the one map for the pre-spawned circus.

File: src/openrct2/ride/Ride.cpp [hunk 4]
```diff
@@ -1802,7 +1805,6 @@ static void ride_music_update(Ride* ride)
             }
         }
     }
```
Comment (Gymnasiast) [inline src/openrct2/ride/Ride.cpp:1807]:
```suggestion
    }

```

File: src/openrct2/ride/Ride.cpp [hunk 5]
```diff
@@ -1754,22 +1754,25 @@ Staff* ride_get_assigned_mechanic(Ride* ride)
  */
 static void ride_music_update(Ride* ride)
 {
-    const auto& rtd = ride->GetRideTypeDescriptor();
-    if (!rtd.HasFlag(RIDE_TYPE_FLAG_MUSIC_ON_DEFAULT) && !rtd.HasFlag(RIDE_TYPE_FLAG_ALLOW_MUSIC))
+    // Circus does not have "music." Music is a sound effect
```
Comment (Gymnasiast) [inline src/openrct2/ride/Ride.cpp:1757]:
```suggestion
    // The circus does not have music in the normal sense - its “music” is a sound effect.
```

**Comment by Gymnasiast** [comment]
Since this does not fully solve the issue, but does improve on it, I decided to revert the changelog and merge this, leaving #15322 open until it is fully fixed. Up until then, at least _some_ circuses will play their music again.

**Review by Gymnasiast — APPROVED** [review]

**Comment by seanmajorpayne** [comment]
Sounds good. I will try to find the source of the other problem.

**Pull request merged**